### DISCUSSION TITLE
sonoff zigbee 3.0 dongle E

### DISCUSSION THREAD

**Discussion by Jacobsentertainment** [Koenkk/zigbee2mqtt#19063] [Devices]
Hi all,

Before I have to read trough the fourth webpage to figure out that there is no clear firmware available for this dongle, would someone here maybe give a good hint where to find the firmware to flash... 

- I have an sonoff ZBDongle 3.0 E (brandnew)
- I want to create an coordinator to run my network with currently about 20 devices. (this will increase most likely)

Our I end up on the Z-stack firmware page [https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator](url) where there is no support for this device, but on the [https://www.zigbee2mqtt.io/guide/adapters/#experimental](url) there is an option but totally not clear where this firmware is avaible....
And I read on the I-tead webpage thats the stick is supported by zigbee2mqtt ea Domoticz. I'm lost here do I need to flash or not? 

Is it me, things need to clear up our am I blind....

Regards Olaf.

**Comment by francisp2** [comment]
https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP

**Comment by Vitus-TL** [comment]
The Sonoff ZBDongle-E comes with the correct coordinator firmware, so no need to flash. In any case, don't use Zigbee (EZSP) 7.3.1.0 build 176 (it doesn't work). Currently I use Zigbee (EZSP) 6.10.3.0 build 297, which works.. From the official documentation:
Coordinator:
https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP
Router:
https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/Router

I used the web-flasher to restore the correct firmware (because of a test with Thread, with MultiPAN RCP-firmware, Zigbee2MQTT doesn't work, Home Assistant's ZHA does... ):
https://dialedin.com.au/blog/sonoff-zbdongle-e-rcp-firmware#web-flasher

**Reply by meelismei** [reply]
@Vitus-TL What exactly doesn't work in firmware 7.3.1.0?
I use Zigbee2MQTT in a zigbee network and everything works fine... but after 34 devices (17 end devices + 17 routers), I can't add any more devices... and most of the end devices (15) are only connected to the coordinator. Could this behavior be what you describe as not working?
Can I downgrade the firmware without having to pair all devices again?

**Comment by metramo** [comment]
Hi Vitus-TL,

Can I interpret that Zigbee2MQTT does not work on this device ?

**Reply by chris-1243** [reply]
Zigbee2mqtt works perfectly fine with Sonoff Dongle-E. The preferred firmware are 7.4.5 or 8.0.2 both with latest `ember` driver. Some firmware are known to have some issue like 7.3.1 that is it...
